webapi/functions/user_functions.py

```diff
@@ -16,19 +19,23 @@ def register_helper(User):
         status_code = 400
         statement = (check_registration_input(username, email, password))
     else:
-        hashed_password = generate_password_hash(request.data['password'], method='sha256')
-
-        user = User.query.filter_by(username=username).first()
-        if not user:
-            user = User(username=username, email=email,
-                        password=hashed_password, public_id=str(uuid.uuid4()),
-                        logged_in=False)
-            user.save()
-            status_code = 201
-            statement = {"message":"Registration successful, log in to access your account"}
+        password_strength, improvements = meter.test(password)
+        if password_strength < 0.5:
+            status_code = 400
+            statement = {"message":"At least 6 characters required for password"}
         else:
-            status_code = 409
-            statement = {"message":"Username or email already registered"}
+            hashed_password = generate_password_hash(request.data['password'], method='sha256')
+            user = User.query.filter_by(username=username).first()
+            if not user:
+                user = User(username=username, email=email,
+                            password=hashed_password, public_id=str(uuid.uuid4()),
+                            logged_in=False)
+                user.save()
+                status_code = 201
+                statement = {"message":"Registration successful, log in to access your account"}
```

Comment:
nice approach of setting the password in login, although you could reduce the if else statements

Reply:
Thanks for the review. I agree,  a good password should contain both letters and numbers but in our case I did not see need for extremely strong passwords. I may add the functionality later on in the API's cycle. Working on ways to reduce my if else statements. The python switch implementation with dictionaries is still a bit challenging for me.